include/circt/Dialect/FIRRTL/FIRRTLAnnotations.h

```diff
@@ -32,6 +32,8 @@ class AnnotationSet {
   /// Get an annotation set for the specified operation.
   explicit AnnotationSet(Operation *op);
 
+  explicit AnnotationSet(ArrayAttr array);
```

Comment:
These are all great improvements to FIRRTLAnnotations, feel free to land them separately if you'd like.